src/librustdoc/html/markdown.rs

```diff
@@ -715,6 +778,53 @@ impl LangString {
                         seen_other_tags = true;
                     }
                 }
+                x if extra.is_some() => {
+                    let s = x.to_lowercase();
+                    match if s == "compile-fail" || s == "compile_fail" || s == "compilefail" {
+                        Some((
+                            "compile_fail",
+                            "the code block will either not be tested if not marked as a rust one \
```

Comment:
Rather than this either/or can we please check if the block would be considered rust and be a little more specific?  If that's a pain then fair enough we can skip it for now.

Reply:
No, I thought about it but wasn't sure. Let's go for it then!

Reply:
This is actually far more complicated than I imagined... I think it'll have to be a PR on its own...

Reply:
OK